### PR TITLE
UnitControl: Add support for complex CSS in the UI

### PR DESCRIPTION
**Pull request by stacimc**
This is an alternative implementation to #38096, which explored adding an actual button to the UnitControl that would toggle open a custom CSS input (as opposed to the dropdown).

## Description

Adds support for a `css` "unit" to the UnitControl which can be used to add complex CSS values like `max...` or `clamp...` through the UI. Themes can configure whether or not to support the unit by its inclusion in the `spacing` settings in `theme.json`:

```
"spacing": {
			"customPadding": true,
			"units": [
				"px",
				"em",
				"rem",
				"vh",
				"vw",
				"css"
			]
		},
```

These types of values are already _technically_ supported by attributes that accept custom units (like padding, margin, and the Spacer block's height), in the sense that you can manually set those attributes (using the Code editor, for example). But since the `UnitControl` doesn't currently support these arbitrary strings, you'll see an empty input in the UI and you won't be able to edit the value through the control panel.

This can lead to strange scenarios. For example, the space beneath the header in Twenty Twenty-Two uses a spacer with height `max(1.25rem, 5vw)` as defined in the theme. If a user goes to edit that spacer, they won't see that value in the Inspector Controls (the height input appears empty). If they set their own value in the controls, they may not realize what they're overriding. It would be nice to be able to quickly see what values are set and tweak them yourself.

## Notes:
* TL;DR I disabled the custom CSS unit if the legacy unit prop is passed.
    * When `value` and `unit` are passed as separate props, for example in the Min Height control for the Cover block, it's usually because those values are stored in separate attributes. In Cover's case, `height` is an attribute with type `number`, and arbitrary custom CSS strings simply can't be stored in it. The best way I can think of handling this is by assuming all uses of the legacy unit prop cannot support custom CSS, and removing the unit even if it is otherwise supported in `theme.json`.
    

## Screenshots 

Video below also has changes from #38340 applied

https://user-images.githubusercontent.com/63313398/151614420-72949cfc-98e9-4e21-b3ec-12d68deb5ba6.mov

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). 
- [ ] I've updated related schemas if appropriate.

### CODE REVIEW

**Comment by stacimc** [comment]
This is working pretty well for a first draft, although it needs much more testing. Unit tests need to be added, and additional storybook examples. Here's a rough breakdown of major concerns:

### Design considerations

* Help text would definitely be useful for the custom CSS unit.
* When used in the BoxControl, particularly when unlinked, the inputs are quite narrow (which can be very apparent/frustrating if you're attempting to add complex values)

### Lack of validation of CSS:

* There’s no visual indication that invalid CSS has been entered. Using the Spacer block as an example, If I type nonsense like `asdf` or just invalid css like `calc(x + anotherVariableThatDoesntExist)` into the `minHeight` input, the block in the editor doesn’t visually update at all — it remains whatever its last valid height was. On the frontend, the invalid css is applied inline and results in a 0-height Spacer.
* While it looks like tags get escaped correctly 😅, it _is_ totally possible to enter arbitrary CSS that has nothing to do with the input. So again using Spacer minHeight, you can select the CSS unit and type: `10px; background-color: orange`. In the editor this is invalid and the block does not visually change, but on the frontend you’ll get a 10px tall orange Spacer!

### Thoughts about `theme.json`:

However this is implemented, I think it’s very important that it should be configurable via `theme.json` so that theme developers can decide whether they want this exposed to users, particularly because it’s a more advanced styling option. In this PR, it’s configurable like any other unit via the `spacing.units` setting, meaning it has to be opted in to.

The problem is that the `UnitControl` supports a legacy api that allows `value` and `unit` to be passed as separate props. This is generally done because they are stored in separate attributes, and usually the value attr has a “number” type. When this happens, custom CSS simply can’t be stored in the relevant attributes... so the `UnitControl` will automatically disable custom CSS if it detects the presence of a legacy `unit` prop.

That’s strange because it creates an unusual scenario where a theme developer can opt into `css` support in `spacing.units`, but the unit won’t actually appear in _all controls_. From a quick look, this includes:

* Cover minHeight
* Search width
* Social link minHeight

### Technical considerations

This approach requires a lot of special-handling for the custom CSS unit, because it's _not really_ a unit in the way that the others are. It shouldn't get saved to attributes or appended to the end of the `value`. It has the disadvantage of muddying the waters about what a unit means within the Control.

### Mobile support

When the design is more stable, we should look into adding mobile support as well.

**Comment by Mamaduka** [comment]
I'm doing some old PR cleanup. Since there has been no activity in the last three years, I'm going to close this as "stale."
